File: src/scanner/code/__fixtures__/javascript/vulnerable.js
/**
 * Test fixture: JavaScript code with quantum-vulnerable cryptography.
 */
const crypto = require('crypto');
const jwt = require('jsonwebtoken');

// RSA key generation (CRITICAL)
const { publicKey, privateKey } = crypto.generateKeyPairSync('rsa', {
  modulusLength: 2048,
  publicKeyEncoding: { type: 'spki', format: 'pem' },
  privateKeyEncoding: { type: 'pkcs8', format: 'pem' },
});

// EC key generation (CRITICAL)
const ecKeys = crypto.generateKeyPairSync('ec', {
  namedCurve: 'P-256',
});

// Ed25519 (CRITICAL)
const edKeys = crypto.generateKeyPairSync('ed25519');

// X25519 key exchange (CRITICAL)
const x25519Keys = crypto.generateKeyPairSync('x25519');

// ECDH (CRITICAL)
const ecdh = crypto.createECDH('secp256k1');
ecdh.generateKeys();

// DH (CRITICAL)
const dh = crypto.createDiffieHellman(2048);

// MD5 hash (CRITICAL)
const md5 = crypto.createHash('md5');

// SHA-1 hash (CRITICAL)
const sha1 = crypto.createHash('sha1');

// AES-128 cipher (MODERATE)
const cipher128 = crypto.createCipheriv('aes-128-gcm', key16, iv);

// JWT with RSA (CRITICAL)
const token = jwt.sign({ sub: '1234' }, privateKey, { algorithm: 'RS256' });

// JWT with ECDSA (CRITICAL)
const ecToken = jwt.sign({ sub: '1234' }, ecPrivateKey, { algorithm: 'ES256' });

File: src/scanner/code/__fixtures__/rust/vulnerable.rs
use rsa::{RsaPrivateKey, RsaPublicKey};
use ecdsa::SigningKey;
use ring::agreement;
use md5::Md5;
use ring::signature;
use digest::Digest;

fn main() {
    let mut rng = rand::thread_rng();

    // RSA key generation (CRITICAL - Shor's algorithm)
    let private_key = RsaPrivateKey::new(&mut rng, 2048).unwrap();
    let public_key = RsaPublicKey::from(&private_key);

    // ECDSA signing key (CRITICAL - Shor's algorithm)
    let signing_key = SigningKey::random(&mut rng);

    // ring ECDH key agreement (CRITICAL - Shor's algorithm)
    let ephemeral_private_key =
        agreement::EphemeralPrivateKey::generate(&agreement::X25519, &rng).unwrap();

    // MD5 hash (CRITICAL - cryptographically broken)
    let hash = Md5::digest(b"data");

    // ring Ed25519 signature (CRITICAL)
    let kp = signature::Ed25519KeyPair::from_pkcs8(pkcs8_bytes).unwrap();

    println!("Private key: {:?}", private_key);
    println!("Signing key: {:?}", signing_key);
    println!("Hash: {:?}", hash);
}
